tools/delete_mri_upload.pl

```diff
@@ -0,0 +1,578 @@
+	# indirectly to the tarchive with ID $tarchiveId
+    my $query = 'SELECT fi.IntermedID, f.FileID, f.File FROM files_intermediary fi '
+        .       'JOIN files f ON (fi.Output_FileID=f.FileID) '
+        .       'WHERE f.SourceFileID IN (SELECT FileID FROM files WHERE TarchiveSource = ?)';
+
+	my $filesRef = $dbh->selectall_arrayref($query, { Slice => {} }, $tarchiveID);
+
+	# Make sure all paths are absolute paths
+    foreach(@$filesRef) {
+		$_->{'File'} = "$dataDirBasePath/$_->{'File'}" unless $_->{'File'} =~ /^\//;
+	} 
+	
+	return $filesRef;   
+}
+
+=pod
+
+=head3 getPicFilesRef($dbh, $tarchiveID, $dataDirBasePath)
+
+Gets the absolute paths of all the files associated to an archive 
+that are listed in table C<parameter_file> and have a parameter
+type set to C<check_pic_filename>.
+
+INPUTS:
+
+  - $dbhr  : database handle reference.
+  - $tarchiveID: ID of the tarchive.
+  - $dataDirBasePath: base path of the directory where all the files in table C<files>
+                      are located (i.e config value of setting 'dataDirBasePath').
+
+RETURNS: 
+
+  an array of hash references. Each hash has two keys: 'FileID' => FileID of a file 
+  in table C<parameter_file> and 'Value' => absolute path of the file with the given ID.
+
+=cut
+sub getPicFilesRef {
+	my($dbh, $tarchiveID, $dataDirBasePath) = @_;
+	
+	# Get all files in parameter_file with parameter_type set to
+	# 'check_pic_filename' that are tied (indirectly) to the tarchive
+	# with ID $tarchiveId
+    (my $query =<<QUERY) =~ s/\s+/ /g;
+        SELECT FileID, Value FROM parameter_file pf
+	    JOIN files f USING (FileID)
+        JOIN parameter_type AS pt
+        ON (pt.ParameterTypeID=pf.ParameterTypeID)
+        WHERE pt.Name='check_pic_filename'
+	    AND (
+	            f.TarchiveSource = ? 
+	         OR f.SourceFileID IN (SELECT FileID FROM files WHERE TarchiveSource = ?)
+	    )
+QUERY
+
+	my $filesRef = $dbh->selectall_arrayref($query, { Slice => {} }, $tarchiveID, $tarchiveID);
+	
+	# Make sure all paths are absolute paths
+	foreach(@$filesRef) {
+		if($_->{'Value'} !~ /^\//) {
+		    $_->{'Value'} = sprintf("%s/%s/%s", $dataDirBasePath, PIC_SUBDIR, $_->{'Value'});
+	    }
+	}
+	
+	return $filesRef;
+}
+
+=pod
+
+=head3 getBackupFileName
+
+Gets the name of the tar compressed file that will contain a backup of all files
+that the script will delete.
+
+
+RETURNS: 
+
+  backup file name.
+
+=cut
+sub getBackupFileName {
+	my($uploadID) = @_;
+	
+	return "mri_upload.$uploadID.tar.gz";
+}
+
+=pod
+
+=head3 backupFiles($uploadId, $archiveLocation, $filesRef, $intermediaryFilesRef, $picFilesRef)
+
+Backs up all the files associated to the archive before deleting them. The backed up files will
+be stored in a C<.tar.gz> archive where all paths are relative to C</> (i.e absolute paths).
+
+INPUTS:
+
+  - $uploadId  : ID of the upload to delete.
+  - $archiveLocation: absolute path of the backed up archive created by the MRI pipeline.
```

Comment:
description of archiveLocation is confusing (the part absolute path of the backed up archive confuses me; isn't it the absolute path to the DICOM archive that needs to be backed up?)

Reply:
I changed the description. I hope it's clear now.